src/assets/scripts/client/aircraft/AircraftCommander.js

```diff
@@ -260,7 +259,7 @@ export default class AircraftCommander {
         aircraft.target.turn = direction;
 
         if (aircraft.hasApproachClearance) {
-            aircraft.cancelApproachClearance(aircraft.altitude, aircraft.heading);
+            aircraft.cancelApproachClearance(aircraft);
```

Comment:
You could probably drop the aircraft param here too and just use`this` when in `.cancelApproachClearance()`

Reply:
Thanks for pointing this out, because it doesn't look right for an entirely different reason: `.cancelApproachClearance()` is not a property of the aircraft at all, but rather the pilot. Fixed that mistake right up!

It is for that reason that we actually do have to pass the aircraft (or at least a couple of its properties) to the pilot, since he does not have access to it for reasons of circularity (possibly a word?).

That's actually a pattern I think would be wise with the pilot rather than passing hundreds of individual properties to the pilot when he completes an action: let him take a look at the entire aircraft, and then manipulate the mcp and fms appropriately. I suppose we could even remove his local references if desired. That's all beyond the scope of this anyway, but food for thought?